pkg/sql/gcjob_test/gcjob_test_test: TestSchemaChangeGCJob failed

pkg/sql/gcjob_test/gcjob_test_test.TestSchemaChangeGCJob [failed](https://tanzanite.cluster.engflow.com/invocation/4cf4e7c1-c4d9-4d1b-a99a-a24f2e381015) on master @ [37ad01a3972cb4d34bfc6dfb4b9cfcac360b15dd](https://github.com/cockroachdb/cockroach/commits/37ad01a3972cb4d34bfc6dfb4b9cfcac360b15dd):

Fatal error:

```
panic: test timed out after 4m57s
running tests:
	TestSchemaChangeGCJob (4m57s)

```
Stack: 

```
goroutine 8537 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2259 +0x3b9
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
* 
* goroutine 2113 [select, 2 minutes]:
* github.com/cockroachdb/pebble/vfs.(*diskHealthCheckingFile).startTicker.func1()
* 	github.com/cockroachdb/pebble/vfs/external/com_github_cockroachdb_pebble/vfs/disk_health.go:171 +0xcb
* created by github.com/cockroachdb/pebble/vfs.(*diskHealthCheckingFile).startTicker in goroutine 1189
* 	github.com/cockroachdb/pebble/vfs/external/com_github_cockroachdb_pebble/vfs/disk_health.go:166 +0x58
* 
* goroutine 2130 [sync.Cond.Wait, 2 minutes]:
* sync.runtime_notifyListWait(0xc005c1a0a0, 0x207)
* 	GOROOT/src/runtime/sema.go:527 +0x159
* sync.(*Cond).Wait(0xc154cf7bf3cca94f?)
* 	GOROOT/src/sync/cond.go:70 +0x85
* github.com/cockroachdb/pebble/record.(*flusherCond).Wait(...)
* 	github.com/cockroachdb/pebble/record/external/com_github_cockroachdb_pebble/record/log_writer.go:204
* github.com/cockroachdb/pebble/record.(*LogWriter).flushLoop(0xc005c1a000, {0x490a800, 0x9bef040})
* 	github.com/cockroachdb/pebble/record/external/com_github_cockroachdb_pebble/record/log_writer.go:438 +0x67c
* runtime/pprof.Do({0x65204d8?, 0x9bef040?}, {{0xc0001febc0?, 0xc001b43920?, 0x0?}}, 0xc003db7fc0)
* 	GOROOT/src/runtime/pprof/runtime.go:51 +0x9d
* github.com/cockroachdb/pebble/record.NewLogWriter.func2()
* 	github.com/cockroachdb/pebble/record/external/com_github_cockroachdb_pebble/record/log_writer.go:357 +0x56
* created by github.com/cockroachdb/pebble/record.NewLogWriter in goroutine 1189
* 	github.com/cockroachdb/pebble/record/external/com_github_cockroachdb_pebble/record/log_writer.go:356 +0x416
* 
* goroutine 7952 [select, 2 minutes]:
* database/sql.(*DB).connectionOpener(0xc002d1dad0, {0x65205b8, 0xc0042775e0})
* 	GOROOT/src/database/sql/sql.go:1218 +0x87
* created by database/sql.OpenDB in goroutine 25
* 	GOROOT/src/database/sql/sql.go:791 +0x165
* 
* goroutine 8397 [chan receive, 2 minutes]:
* pkg/sql/gcjob_test/gcjob_test_test_test.doTestSchemaChangeGCJob.func1(0x65205f0?)
* 	pkg/sql/gcjob_test/gcjob_test_test_test/pkg/sql/gcjob_test/gc_job_test.go:92 +0x19
* github.com/cockroachdb/cockroach/pkg/sql/gcjob.schemaChangeGCResumer.deleteDataAndWaitForGC({_}, {_, _}, {0xc00188c000, {0xc0010dcfc0, 0xc001be7f40, 0xc0010dcfb0, 0xc002f984e0}, {{0xc0013ba7e0, 0xc0013ba7f8}, ...}, ...}, ...)
* 	github.com/cockroachdb/cockroach/pkg/sql/gcjob/gc_job.go:325 +0xc8
* github.com/cockroachdb/cockroach/pkg/sql/gcjob.schemaChangeGCResumer.Resume({0xc0047e0340}, {0x65205f0, 0xc005223aa0}, {0x4defd80?, 0xc0043713c0})
* 	github.com/cockroachdb/cockroach/pkg/sql/gcjob/gc_job.go:313 +0x2b9
* github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine.func2(0xc0047e0340?, 0x0?, 0xc00330b848, {0x6508290?, 0xc0043712f8?}, {0x65205f0?, 0xc005223aa0?}, {0x4defd80?, 0xc0043713c0?})
* 	github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1589 +0xc2
* github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine(0xc0001fd900, {0x65205f0?, 0xc005223950}, {0x4defd80, 0xc0043713c0}, {0x6508290, 0xc0043712f8}, 0xc0047e0340, {0x5121b89, 0x7}, ...)
* 	github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1590 +0xa2d
* github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).runJob(0xc0001fd900, {0x65205f0, 0xc005223950}, {0x6508290, 0xc0043712f8}, 0xc0047e0340, {0x5121b89, 0x7}, {0xc0014b0978, 0x16})
* 	github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:434 +0x4c7
* github.com/cockroachdb/cockroach/pkg/jobs.(*StartableJob).Start.func2({0xc00416fc00?, 0x3924e85?})
* 	github.com/cockroachdb/cockroach/pkg/jobs/jobs.go:924 +0xe7
* github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2()
* 	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:484 +0x13a
* created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 25
* 	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:475 +0x415
* 
*
```
</p>
</details>
<p>Parameters: <code>=</code>
, <code>attempt=1</code>
, <code>run=2</code>
, <code>shard=1</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSchemaChangeGCJob.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-34254

pkg/sql/gcjob_test/gcjob_test_test.TestSchemaChangeGCJob [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/13090583?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/13090583?buildTab=artifacts#/) on master @ [83bcafa004f79bdb156f6921fb8e898df9925c08](https://github.com/cockroachdb/cockroach/commits/83bcafa004f79bdb156f6921fb8e898df9925c08):

Fatal error:

```
panic: test timed out after 4m57s
running tests:
	TestSchemaChangeGCJob (4m57s)

```
Stack: 

```
goroutine 124166 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2259 +0x320
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
* created by github.com/cockroachdb/cockroach/pkg/util/admission.(*WorkQueue).startClosingEpochs in goroutine 62
* 	github.com/cockroachdb/cockroach/pkg/util/admission/work_queue.go:473 +0x5c
* 
* goroutine 114650 [chan receive, 2 minutes]:
* github.com/cockroachdb/pebble.(*tableCacheShard).releaseLoop.func1({0x5eae7e8, 0x400e5739b0})
* 	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_cache.go:392 +0x80
* runtime/pprof.Do({0x5eae698?, 0x90c7d60?}, {{0x4000d9bd80?, 0x400761ad30?, 0x0?}}, 0x4001cf0f98)
* 	GOROOT/src/runtime/pprof/runtime.go:51 +0x80
* github.com/cockroachdb/pebble.(*tableCacheShard).releaseLoop(0x4001cf0f58?)
* 	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_cache.go:390 +0x5c
* created by github.com/cockroachdb/pebble.(*tableCacheShard).init in goroutine 62
* 	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_cache.go:382 +0xe4
* 
* goroutine 114661 [chan receive, 2 minutes]:
* github.com/cockroachdb/pebble.(*tableCacheShard).releaseLoop.func1({0x5eae7e8, 0x40014aea50})
* 	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_cache.go:392 +0x80
* runtime/pprof.Do({0x5eae698?, 0x90c7d60?}, {{0x4000d9bd80?, 0x40012ea7c0?, 0x0?}}, 0x4001d4e798)
* 	GOROOT/src/runtime/pprof/runtime.go:51 +0x80
* github.com/cockroachdb/pebble.(*tableCacheShard).releaseLoop(0x4001d4e758?)
* 	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_cache.go:390 +0x5c
* created by github.com/cockroachdb/pebble.(*tableCacheShard).init in goroutine 62
* 	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_cache.go:382 +0xe4
* 
* goroutine 114668 [chan receive, 2 minutes]:
* github.com/cockroachdb/pebble.(*tableCacheShard).releaseLoop.func1({0x5eae7e8, 0x400b5ace40})
* 	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_cache.go:392 +0x80
* runtime/pprof.Do({0x5eae698?, 0x90c7d60?}, {{0x4000d9bd80?, 0x400631f140?, 0x0?}}, 0x4001d4c798)
* 	GOROOT/src/runtime/pprof/runtime.go:51 +0x80
* github.com/cockroachdb/pebble.(*tableCacheShard).releaseLoop(0x4001d4c758?)
* 	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_cache.go:390 +0x5c
* created by github.com/cockroachdb/pebble.(*tableCacheShard).init in goroutine 62
* 	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_cache.go:382 +0xe4
* 
* goroutine 115162 [select, 2 minutes]:
* github.com/cockroachdb/cockroach/pkg/util/admission.initWorkQueue.func2()
* 	github.com/cockroachdb/cockroach/pkg/util/admission/work_queue.go:425 +0x7c
* created by github.com/cockroachdb/cockroach/pkg/util/admission.initWorkQueue in goroutine 62
* 	github.com/cockroachdb/cockroach/pkg/util/admission/work_queue.go:422 +0x32c
* 
* goroutine 114663 [chan receive, 2 minutes]:
* github.com/cockroachdb/pebble.(*tableCacheShard).releaseLoop.func1({0x5eae7e8, 0x400c4d1f50})
* 	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_cache.go:392 +0x80
* runtime/pprof.Do({0x5eae698?, 0x90c7d60?}, {{0x4000d9bd80?, 0x40001313b0?, 0x0?}}, 0x4001d4af98)
* 	GOROOT/src/runtime/pprof/runtime.go:51 +0x80
* github.com/cockroachdb/pebble.(*tableCacheShard).releaseLoop(0x4001d4af58?)
* 	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_cache.go:390 +0x5c
* created by github.com/cockroachdb/pebble.(*tableCacheShard).init in goroutine 62
* 	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_cache.go:382 +0xe4
* 
*
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSchemaChangeGCJob.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/sql/gcjob_test/gcjob_test_test.TestSchemaChangeGCJob [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/13106887?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/13106887?buildTab=artifacts#/) on master @ [d8ecad995859121df24fb905b6514f0fef6a4271](https://github.com/cockroachdb/cockroach/commits/d8ecad995859121df24fb905b6514f0fef6a4271):

Fatal error:

```
panic: test timed out after 4m57s
running tests:
	TestSchemaChangeGCJob (4m57s)

```
Stack: 

```
goroutine 16318 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2259 +0x320
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
* github.com/cockroachdb/cockroach/pkg/util/admission.(*tokenAllocationTicker).tick(...)
* 	github.com/cockroachdb/cockroach/pkg/util/admission/io_load_listener.go:418
* github.com/cockroachdb/cockroach/pkg/util/admission.(*StoreGrantCoordinators).SetPebbleMetricsProvider.func1()
* 	github.com/cockroachdb/cockroach/pkg/util/admission/grant_coordinator.go:115 +0x88
* created by github.com/cockroachdb/cockroach/pkg/util/admission.(*StoreGrantCoordinators).SetPebbleMetricsProvider in goroutine 25
* 	github.com/cockroachdb/cockroach/pkg/util/admission/grant_coordinator.go:109 +0x264
* 
* goroutine 15757 [chan receive, 2 minutes]:
* pkg/sql/gcjob_test/gcjob_test_test_test.doTestSchemaChangeGCJob.func1(0x5eaf528?)
* 	pkg/sql/gcjob_test/gcjob_test_test_test/pkg/sql/gcjob_test/gc_job_test.go:92 +0x24
* github.com/cockroachdb/cockroach/pkg/sql/gcjob.schemaChangeGCResumer.deleteDataAndWaitForGC({_}, {_, _}, {0x4001f2c000, {0x4001941910, 0x4001eac9a0, 0x4001941900, 0x400208c880}, {{0x4001b75800, 0x4001b75818}, ...}, ...}, ...)
* 	github.com/cockroachdb/cockroach/pkg/sql/gcjob/gc_job.go:325 +0xc0
* github.com/cockroachdb/cockroach/pkg/sql/gcjob.schemaChangeGCResumer.Resume({0x40033cfa00}, {0x5eaf528, 0x400620c990}, {0x477b340?, 0x40064b0bc8})
* 	github.com/cockroachdb/cockroach/pkg/sql/gcjob/gc_job.go:313 +0x1f0
* github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine.func2(0x40033cfa00?, 0x0?, 0x4004bf7830, {0x5e97090?, 0x40069982e8?}, {0x5eaf528?, 0x400620c990?}, {0x477b340?, 0x40064b0bc8?})
* 	github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1589 +0xec
* github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine(0x4002172000, {0x5eaf528?, 0x400620c8d0}, {0x477b340, 0x40064b0bc8}, {0x5e97090, 0x40069982e8}, 0x40033cfa00, {0x4aad06b, 0x7}, ...)
* 	github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1590 +0x78c
* github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).runJob(0x4002172000, {0x5eaf528, 0x400620c8d0}, {0x5e97090, 0x40069982e8}, 0x40033cfa00, {0x4aad06b, 0x7}, {0x4005d580c0, 0x16})
* 	github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:434 +0x370
* github.com/cockroachdb/cockroach/pkg/jobs.(*StartableJob).Start.func2({0x4003d42800?, 0x4005f64930?})
* 	github.com/cockroachdb/cockroach/pkg/jobs/jobs.go:924 +0xdc
* github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2()
* 	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:484 +0x128
* created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 25
* 	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:475 +0x330
* 
* goroutine 2341 [sync.Cond.Wait, 2 minutes]:
* sync.runtime_notifyListWait(0x40064e60a0, 0x26b)
* 	GOROOT/src/runtime/sema.go:527 +0x154
* sync.(*Cond).Wait(0x40064e6090)
* 	GOROOT/src/sync/cond.go:70 +0xcc
* github.com/cockroachdb/pebble/record.(*flusherCond).Wait(...)
* 	github.com/cockroachdb/pebble/record/external/com_github_cockroachdb_pebble/record/log_writer.go:204
* github.com/cockroachdb/pebble/record.(*LogWriter).flushLoop(0x40064e6000, {0x4295c20, 0x90d7d60})
* 	github.com/cockroachdb/pebble/record/external/com_github_cockroachdb_pebble/record/log_writer.go:438 +0x618
* runtime/pprof.Do({0x5eaf3d8?, 0x90d7d60?}, {{0x4000d838a0?, 0x4001b94510?, 0x0?}}, 0x4000e1dfb8)
* 	GOROOT/src/runtime/pprof/runtime.go:51 +0x80
* github.com/cockroachdb/pebble/record.NewLogWriter.func2()
* 	github.com/cockroachdb/pebble/record/external/com_github_cockroachdb_pebble/record/log_writer.go:357 +0x60
* created by github.com/cockroachdb/pebble/record.NewLogWriter in goroutine 1782
* 	github.com/cockroachdb/pebble/record/external/com_github_cockroachdb_pebble/record/log_writer.go:356 +0x3bc
* 
* goroutine 15546 [select, 2 minutes]:
* database/sql.(*DB).connectionOpener(0x4004c00f70, {0x5eaf4f0, 0x4004f796d0})
* 	GOROOT/src/database/sql/sql.go:1218 +0x80
* created by database/sql.OpenDB in goroutine 25
* 	GOROOT/src/database/sql/sql.go:791 +0x158
* 
*
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSchemaChangeGCJob.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/sql/gcjob_test/gcjob_test_test.TestSchemaChangeGCJob [failed](https://tanzanite.cluster.engflow.com/invocations/default/628f4b1e-e8b7-41f6-a501-bf15569cbdae?testReportRun=29&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL2djam9iX3Rlc3Q6Z2Nqb2JfdGVzdF90ZXN0) on master @ [1f53a61732c6f9e48622f45d276eb5e1c343ad30](https://github.com/cockroachdb/cockroach/commits/1f53a61732c6f9e48622f45d276eb5e1c343ad30):

Fatal error:

```
panic: test timed out after 4m57s
running tests:
	TestSchemaChangeGCJob (4m57s)

```
Stack: 

```
goroutine 92247 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2259 +0x3b9
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
* github.com/cockroachdb/pebble.openCleanupManager.func1.1({0x65863f0, 0xc005575260})
* 	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/cleaner.go:86 +0x17
* runtime/pprof.Do({0x65862d8?, 0x9cf0180?}, {{0xc000d4d100?, 0x12af4e0?, 0xc00444f340?}}, 0xc003084fc0)
* 	GOROOT/src/runtime/pprof/runtime.go:51 +0x9d
* github.com/cockroachdb/pebble.openCleanupManager.func1()
* 	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/cleaner.go:85 +0x56
* created by github.com/cockroachdb/pebble.openCleanupManager in goroutine 8
* 	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/cleaner.go:84 +0x298
* 
* goroutine 83353 [select, 2 minutes]:
* github.com/cockroachdb/cockroach/pkg/util/admission.initWorkQueue.func2()
* 	github.com/cockroachdb/cockroach/pkg/util/admission/work_queue.go:425 +0x85
* created by github.com/cockroachdb/cockroach/pkg/util/admission.initWorkQueue in goroutine 8
* 	github.com/cockroachdb/cockroach/pkg/util/admission/work_queue.go:422 +0x3ce
* 
* goroutine 83326 [chan receive, 2 minutes]:
* github.com/cockroachdb/pebble.(*tableCacheShard).releaseLoop.func1({0x65863f0, 0xc005575320})
* 	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_cache.go:392 +0x94
* runtime/pprof.Do({0x65862d8?, 0x9cf0180?}, {{0xc000d4d140?, 0x65862d8?, 0xc001ffdfd0?}}, 0xc001ffdfa8)
* 	GOROOT/src/runtime/pprof/runtime.go:51 +0x9d
* github.com/cockroachdb/pebble.(*tableCacheShard).releaseLoop(0xc0016b0000?)
* 	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_cache.go:390 +0x52
* created by github.com/cockroachdb/pebble.(*tableCacheShard).init in goroutine 8
* 	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_cache.go:382 +0xe5
* 
* goroutine 83354 [select]:
* github.com/cockroachdb/cockroach/pkg/util/admission.(*WorkQueue).startClosingEpochs.func1()
* 	github.com/cockroachdb/cockroach/pkg/util/admission/work_queue.go:502 +0x14d
* created by github.com/cockroachdb/cockroach/pkg/util/admission.(*WorkQueue).startClosingEpochs in goroutine 8
* 	github.com/cockroachdb/cockroach/pkg/util/admission/work_queue.go:473 +0x4f
* 
* goroutine 83487 [select, 2 minutes]:
* github.com/cockroachdb/cockroach/pkg/util/admission.makeStoreWorkQueue.func1()
* 	github.com/cockroachdb/cockroach/pkg/util/admission/work_queue.go:2241 +0x85
* created by github.com/cockroachdb/cockroach/pkg/util/admission.makeStoreWorkQueue in goroutine 8
* 	github.com/cockroachdb/cockroach/pkg/util/admission/work_queue.go:2238 +0x4b1
* 
* goroutine 83363 [chan receive, 2 minutes]:
* github.com/cockroachdb/pebble.(*cleanupManager).mainLoop(0xc003f3df40)
* 	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/cleaner.go:155 +0xdb
* github.com/cockroachdb/pebble.openCleanupManager.func1.1({0x65863f0, 0xc005575410})
* 	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/cleaner.go:86 +0x17
* runtime/pprof.Do({0x65862d8?, 0x9cf0180?}, {{0xc000d4d100?, 0xc0000b3fd0?, 0x64f288?}}, 0xc0000b3fc0)
* 	GOROOT/src/runtime/pprof/runtime.go:51 +0x9d
* github.com/cockroachdb/pebble.openCleanupManager.func1()
* 	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/cleaner.go:85 +0x56
* created by github.com/cockroachdb/pebble.openCleanupManager in goroutine 8
* 	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/cleaner.go:84 +0x298
* 
*
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>run=29</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSchemaChangeGCJob.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/gcjob_test/gc_job_test.go
+++ b/pkg/sql/gcjob_test/gc_job_test.go
@@ -74,16 +74,19 @@ func TestSchemaChangeGCJob(t *testing.T) {
 
 	for _, dropItem := range []DropItem{INDEX, TABLE, DATABASE} {
 		for _, ttlTime := range []TTLTime{PAST, SOON, FUTURE} {
-			// NB: The inner body of this loop has been extracted into a function to
-			// ensure defer statements (namely testserver.Stop) is executed at the
-			// end of each loop rather than the end of each test.
-			doTestSchemaChangeGCJob(t, dropItem, ttlTime)
+			t.Run(fmt.Sprintf("dropItem=%d/ttlTime=%d", dropItem, ttlTime), func(t *testing.T) {
+				// NB: The inner body of this loop has been extracted into a function to
+				// ensure defer statements (namely testserver.Stop) is executed at the
+				// end of each loop rather than the end of each test.
+				doTestSchemaChangeGCJob(t, dropItem, ttlTime)
+			})
 		}
 	}
 }
 
 func doTestSchemaChangeGCJob(t *testing.T, dropItem DropItem, ttlTime TTLTime) {
 	blockGC := make(chan struct{}, 1)
+	defer close(blockGC)
 	params := base.TestServerArgs{}
 	params.ScanMaxIdleTime = time.Millisecond
 	params.Knobs.JobsTestingKnobs = jobs.NewTestingKnobsWithShortIntervals()
@@ -227,9 +230,11 @@ func doTestSchemaChangeGCJob(t *testing.T, dropItem DropItem, ttlTime TTLTime) {
 
 	// Check that the job started.
 	jobIDStr := strconv.Itoa(int(job.ID()))
-	if err := jobutils.VerifyRunningSystemJob(t, sqlDB, 0, jobspb.TypeSchemaChangeGC, sql.RunningStatusWaitingGC, lookupJR); err != nil {
-		t.Fatal(err)
-	}
+	testutils.SucceedsSoon(t, func() error {
+		return jobutils.VerifyRunningSystemJob(
+			t, sqlDB, 0, jobspb.TypeSchemaChangeGC, sql.RunningStatusWaitingGC, lookupJR,
+		)
+	})
 
 	if ttlTime != FUTURE {
 		// Check that the job eventually blocks right before performing GC, due to the testing knob.

